### PR TITLE
(SERVER-1303) Adds method that merges TK Auth rules into existing config

### PR DESCRIPTION
**Pull request by erikPrime**
This is a continued effort from PR #1046.  @kurtwall made some comments, and I made some minor updates to those comments.

### CODE REVIEW

File: acceptance/lib/helper.rb [hunk 1]
```diff
@@ -239,6 +239,74 @@ def https_request(url, request_method, cert, key, body = nil)
...
+    #have an invalid match-request.
+    match_request.merge!('allow' => allow) if allow
+    match_request.merge!('allow-unauthenticated' => true) if allow_unauthenticated
+    match_request.merge!('deny' => deny) if deny
+
+    authconf_text = on(master, "cat #{authconf_file}").stdout
+    authconf_hash = Hocon.parse(authconf_text)
+    authconf_hash['authorization']['rules'] << match_request
+ 
+    modify_tk_config(host, authconf_file, authconf_hash, true)
```
Comment (erikPrime) [inline acceptance/lib/helper.rb]: Yes, this is a beaker method; if during review, we decide we want to do it our own way with hocon, I can do that.

File: acceptance/lib/helper.rb [hunk 2]
```diff
@@ -239,6 +239,74 @@ def https_request(url, request_method, cert, key, body = nil)
...
+    allow_unauthenticated = args[:allow_unauthenticated] || false
+    deny          	  = args[:deny] || false
+    sort_order   	  = args[:sort_order] || 77
+    path          	  = args[:path] || '/'
+    type     		  = args[:type] || 'path'
+    default_http_methods  = ['head', 'get', 'put', 'post', 'delete']
+    method        	  = args[:method] || default_http_methods
+    query_params  	  = args[:query_params] || {}
+    #TODO: handle TK-293 X509 extensions.
+    authconf_file 	  = args[:authconf_file] ||
```
Comment (jpinsonault) [inline acceptance/lib/helper.rb]: I don't know how much you or anyone else cares, but the spaces on the `=`s is off on some of these vars

Comment (erikPrime) [inline acceptance/lib/helper.rb]: Fixing...

File: acceptance/lib/helper.rb (hunk 1 above)
Comment (jpinsonault) [inline acceptance/lib/helper.rb]: If it works, it works. Looks like it just dumps json. Im assuming we don't care about any comments that might be in there during an acceptance test. Not that a general hocon file modifying helper wouldn't be good.

**Comment by jpinsonault** [comment]
👍

**Comment by kurtwall** [comment]
👍

**Pull request merged**